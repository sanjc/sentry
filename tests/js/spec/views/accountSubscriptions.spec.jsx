import PropTypes from 'prop-types';
import React from 'react';
import {shallow, mount} from 'enzyme';

import {Client} from 'app/api';
import AccountSubscriptions from 'app/views/settings/account/accountSubscriptions';

describe('AccountSubscriptions', function() {
  beforeEach(function() {
    Client.clearMockResponses();
  });

  it('renders empty', function() {
    Client.addMockResponse({
      url: '/account/subscriptions/',
      body: [],
    });
    let wrapper = shallow(<AccountSubscriptions />, {
      context: {
        router: TestStubs.router(),
      },
      childContextTypes: {
        router: PropTypes.object,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders list and can toggle', function() {
    Client.addMockResponse({
      url: '/account/subscriptions/',
      body: TestStubs.Subscriptions(),
    });

    let wrapper = mount(<AccountSubscriptions />, {
      context: {
        router: TestStubs.router(),
      },
      childContextTypes: {
        router: PropTypes.object,
      },
    });

    expect(wrapper).toMatchSnapshot();

    let subscribeResp = {
      url: '/account/subscriptions/',
      method: 'PUT',
      data: {
        list_id: 'test list id',
        subscribed: true,
      },
    };

    Client.addMockResponse(subscribeResp);

    expect(Client.getCallCount(subscribeResp)).toBe(0);

    wrapper
      .find('Switch')
      .first()
      .simulate('click');

    expect(Client.getCallCount(subscribeResp)).toBe(1);
  });
});
