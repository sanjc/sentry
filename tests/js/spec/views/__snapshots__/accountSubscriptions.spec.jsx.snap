// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSubscriptions renders empty 1`] = `
<DocumentTitle
  title="Sentry"
>
  <div>
    <SettingsPageHeading
      label="Subscriptions"
    />
    <p>
      Sentry is committed to respecting your inbox. Our goal is to provide useful content and resources that make fixing errors less painful. Enjoyable even.
    </p>
    <p>
      As part of our compliance with the EU’s General Data Protection Regulation (GDPR), starting on 25 May 2018, we’ll only email you according to the marketing categories to which you’ve explicitly opted-in.
    </p>
    <Styled(div)>
      <PanelHeader
        disablePadding={true}
      >
        <Flex
          align="center"
        >
          <Box
            flex="1"
            px={2}
          >
            Name
          </Box>
          <Box
            px={2}
          >
            Subscribed
          </Box>
        </Flex>
      </PanelHeader>
      <Styled(div) />
    </Styled(div)>
    <p>
      We’re applying GDPR consent and privacy policies to all Sentry contacts, regardless of location. You’ll be able to manage your subscriptions here and from an Unsubscribe link in the footer of all marketing emails.
    </p>
    <p>
      Please contact 
      <a
        href="mailto:learn@sentry.io"
      >
        learn@sentry.io
      </a>
       with any questions or suggestions.
    </p>
  </div>
</DocumentTitle>
`;

exports[`AccountSubscriptions renders list and can toggle 1`] = `
.glamor-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  font-size: 14px;
  box-shadow: inset 0 -1px 0;
  margin-bottom: 30px;
}

.glamor-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  border-bottom: 3px solid;
}

.glamor-0 {
  padding: 0 0 15px;
}

.glamor-46 {
  background: #fff;
  border: 1px solid;
  margin-bottom: 30px;
  position: relative;
}

.glamor-16 {
  border-bottom: 1px solid;
  border-radius: 0 0;
  background: padding:15px 0;
  text-transform: uppercase;
  font-size: 13px;
}

.glamor-14 {
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  margin-bottom: 20px;
  font-size: inherit;
  text-transform: inherit;
  margin: 0;
}

.glamor-12 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.glamor-8 {
  box-sizing: border-box;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-10 {
  box-sizing: border-box;
  padding-left: 16px;
  padding-right: 16px;
}

.glamor-28 {
  box-sizing: border-box;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid;
}

.glamor-28:last-child {
  border: 0;
}

.glamor-22 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-18 {
  font-size: 1.2em;
}

.glamor-20 {
  font-size: 0.8em;
  margin-top: 6px;
}

.glamor-26 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.glamor-24 {
  font-size: 0.8em;
  margin-top: 6px;
  text-align: right;
}

<AccountSubscriptions>
  <DocumentTitle
    title="Sentry"
  >
    <div>
      <SettingsPageHeading
        label="Subscriptions"
      >
        <Styled(div)>
          <div
            className="glamor-6 glamor-7"
          >
            <Styled(div)>
              <div
                className="glamor-4 glamor-5"
              >
                <Styled(div)>
                  <div
                    className="glamor-2 glamor-3"
                  >
                    <Styled(span)>
                      <span
                        className="glamor-0 glamor-1"
                      >
                        Subscriptions
                      </span>
                    </Styled(span)>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
      </SettingsPageHeading>
      <p>
        Sentry is committed to respecting your inbox. Our goal is to provide useful content and resources that make fixing errors less painful. Enjoyable even.
      </p>
      <p>
        As part of our compliance with the EU’s General Data Protection Regulation (GDPR), starting on 25 May 2018, we’ll only email you according to the marketing categories to which you’ve explicitly opted-in.
      </p>
      <Styled(div)>
        <div
          className="glamor-46 glamor-47"
        >
          <PanelHeader
            disablePadding={true}
          >
            <Styled(div)
              disablePadding={true}
            >
              <div
                className="glamor-16 glamor-17"
              >
                <Styled(div)>
                  <div
                    className="glamor-14 glamor-15"
                  >
                    <Flex
                      align="center"
                    >
                      <Base
                        align="center"
                        className="glamor-12"
                      >
                        <div
                          className="glamor-12"
                          is={null}
                        >
                          <Box
                            flex="1"
                            px={2}
                          >
                            <Base
                              className="glamor-8"
                              flex="1"
                              px={2}
                            >
                              <div
                                className="glamor-8"
                                is={null}
                              >
                                Name
                              </div>
                            </Base>
                          </Box>
                          <Box
                            px={2}
                          >
                            <Base
                              className="glamor-10"
                              px={2}
                            >
                              <div
                                className="glamor-10"
                                is={null}
                              >
                                Subscribed
                              </div>
                            </Base>
                          </Box>
                        </div>
                      </Base>
                    </Flex>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
          </PanelHeader>
          <Styled(div)>
            <div
              className="glamor-44 glamor-45"
            >
              <Styled(Base)
                align="center"
                key="2"
                p={2}
              >
                <Base
                  align="center"
                  className="glamor-28 glamor-29"
                  p={2}
                >
                  <div
                    className="glamor-28 glamor-29"
                    is={null}
                  >
                    <Box
                      flex="1"
                    >
                      <Base
                        className="glamor-22"
                        flex="1"
                      >
                        <div
                          className="glamor-22"
                          is={null}
                        >
                          <Styled(div)>
                            <div
                              className="glamor-18 glamor-19"
                            >
                              Product & Feature Updates
                            </div>
                          </Styled(div)>
                          <Styled(div)>
                            <div
                              className="glamor-20 glamor-21"
                            >
                              Everything you need to know about Sentry features, integrations, partnerships, and launches.
                            </div>
                          </Styled(div)>
                        </div>
                      </Base>
                    </Box>
                    <Flex
                      align="flex-end"
                      direction="column"
                    >
                      <Base
                        align="flex-end"
                        className="glamor-26"
                        direction="column"
                      >
                        <div
                          className="glamor-26"
                          is={null}
                        >
                          <Switch
                            isActive={true}
                            size="lg"
                            toggle={[Function]}
                          >
                            <div
                              aria-checked={true}
                              className="switch switch-lg switch-on"
                              onClick={[Function]}
                              role="checkbox"
                            >
                              <span
                                className="switch-toggle"
                              />
                            </div>
                          </Switch>
                          <Styled(div)>
                            <div
                              className="glamor-24 glamor-25"
                            >
                              <div>
                                test@sentry.io
                                 on 
                              </div>
                              <div>
                                <DateTime
                                  date="2018-01-08T05:14:59.102Z"
                                  seconds={true}
                                  shortDate={true}
                                >
                                  <time>
                                    01/08/2018
                                  </time>
                                </DateTime>
                              </div>
                            </div>
                          </Styled(div)>
                        </div>
                      </Base>
                    </Flex>
                  </div>
                </Base>
              </Styled(Base)>
              <Styled(Base)
                align="center"
                key="1"
                p={2}
              >
                <Base
                  align="center"
                  className="glamor-28 glamor-29"
                  p={2}
                >
                  <div
                    className="glamor-28 glamor-29"
                    is={null}
                  >
                    <Box
                      flex="1"
                    >
                      <Base
                        className="glamor-22"
                        flex="1"
                      >
                        <div
                          className="glamor-22"
                          is={null}
                        >
                          <Styled(div)>
                            <div
                              className="glamor-18 glamor-19"
                            >
                              Sentry Newsletter
                            </div>
                          </Styled(div)>
                          <Styled(div)>
                            <div
                              className="glamor-20 glamor-21"
                            >
                              Our monthly update on what's new with Sentry and the community.
                            </div>
                          </Styled(div)>
                        </div>
                      </Base>
                    </Box>
                    <Flex
                      align="flex-end"
                      direction="column"
                    >
                      <Base
                        align="flex-end"
                        className="glamor-26"
                        direction="column"
                      >
                        <div
                          className="glamor-26"
                          is={null}
                        >
                          <Switch
                            isActive={false}
                            size="lg"
                            toggle={[Function]}
                          >
                            <div
                              aria-checked={false}
                              className="switch switch-lg"
                              onClick={[Function]}
                              role="checkbox"
                            >
                              <span
                                className="switch-toggle"
                              />
                            </div>
                          </Switch>
                        </div>
                      </Base>
                    </Flex>
                  </div>
                </Base>
              </Styled(Base)>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
      <p>
        We’re applying GDPR consent and privacy policies to all Sentry contacts, regardless of location. You’ll be able to manage your subscriptions here and from an Unsubscribe link in the footer of all marketing emails.
      </p>
      <p>
        Please contact 
        <a
          href="mailto:learn@sentry.io"
        >
          learn@sentry.io
        </a>
         with any questions or suggestions.
      </p>
    </div>
  </DocumentTitle>
</AccountSubscriptions>
`;
